### ISSUE TITLE
SCM hangs trying to see details of big SVN commit

### ISSUE THREAD

**Issue by markkrj** [scm-manager/scm-manager#1649]
## Issue description

Clicking on `details` in a big SVN commit makes the browser hang and start draining memory.

![image](https://user-images.githubusercontent.com/13537652/117298189-7561f180-ae4d-11eb-8b90-f3c3287ab3f5.png)

#### Describe how to reproduce the bug

Create a subversion repository and commit an entire project tree, then try clicking on the commit details.

#### Describe your SCM-Manager setup

Clean SCM-Manager install on Ubuntu using APT.

#### Which operating system, browser and versions do you use?

SCM-Manager: 2.18.0
Ubuntu: 20.04

I think this has something to do with SCM trying to expand all the diffs in the commit. Imo, it would be best if it just showed modified file names (or even just some files, in case of big commit), so we can click on files we want to see the diff.

**Comment by pfeuffer** [comment]
You're right. This is no issue of SVN but of the big diff. One can get the same result with Mercurial. For Git we introduced limited diffs some releases ago. Looks like we need something like this for SVN and Mercurial, too, to prevent this.

**Comment by markkrj** [comment]
Also, as branches/tags in svn are copies of trunk, I think it would be better to show it in SCM Manager as it shows in `svn log -v`, like:
```
r2 | marcos | 2021-05-06 22:28:45 -0300 (qui, 06 mai 2021) | 1 line
Changed paths:
   A /tags/r1 (from /trunk:1)
```
So we know that it is a copy of trunk in revision 1. No need to show diffs in this case.

**Comment by markkrj** [comment]
Bad bot

**Comment by mcote-57** [comment]
Is this going to be fixed?
We are running into this issue I believe.  It is causing trouble with code reviews in crucible/fisheye.  They are not able to display diffs.  It just times out.  When I try to open the commit in scm-manager web it gives an out of memory error.

**Comment by sdorra** [comment]
@mcote-57 this issue is about the freeze of the browser, which tries to render a large commit. The problem is probably caused by the SyntaxHighlighter, which is blocking the main thread of the browser. We are working on this problem right now.

But this could not cause problems with crucible/fisheye, because they don't use the web interface.

**Comment by mcote-57** [comment]
the problem with fisheye is this one: https://support.tmatesoft.com/t/svnkit-does-not-properly-urlencode-urls-in-outbound-http-requests-svnkit-jetty/2426/9
